tuf/tuf.go

```diff
@@ -845,28 +845,31 @@ func (tr *Repo) SignRoot(expires time.Time) (*data.Signed, error) {
 		}
 	}
 
-	// if the root role has changed and original role had not been saved as a previous role, save it now
-	if !tr.originalRootRole.Equals(currRoot) && !tr.originalRootRole.Equals(latestSavedRole) {
-		rolesToSignWith = append(rolesToSignWith, tr.originalRootRole)
-		latestSavedRole = tr.originalRootRole
-
-		versionName := oldRootVersionName(tempRoot.Signed.Version)
-		tempRoot.Signed.Roles[versionName] = &data.RootRole{
-			KeyIDs: latestSavedRole.ListKeyIDs(), Threshold: latestSavedRole.Threshold}
-
-	}
-
 	tempRoot.Signed.Expires = expires
 	tempRoot.Signed.Version++
 
-	// if the current role doesn't match with the latest saved role, save it
-	if !currRoot.Equals(latestSavedRole) {
-		rolesToSignWith = append(rolesToSignWith, currRoot)
+	// if the root role (root keys or root threshold) has changed, save the
+	// current  role if it hasn't already been saved, as well as saving the
+	// new root role
+	if !tr.originalRootRole.Equals(currRoot) {
+		// if the current role has not been saved before, save it now as
+		// role name "root.<previous version>"
```

Comment:
With the piece of code below to always save the new role, this basically becomes a special case for the initial root role. Would it make sense to remove the below code and just use this piece to always save a root role when it's rotated out? That will also give a consistent meaning to the `<version>`, i.e. it will always be the last version for which that root role was current.